packages/http-specs/specs/payload/pageable/mockapi.ts

```diff
@@ -38,12 +44,243 @@ Scenarios.Payload_Pageable_ServerDrivenPagination_link = passOnSuccess([
     response: {
       status: 200,
       body: json({
-        pets: [
-          { id: "3", name: "bird" },
-          { id: "4", name: "fish" },
-        ],
+        pets: SecondPage,
       }),
     },
     kind: "MockApiDefinition",
   },
 ]);
+
+function reqBodyResBodyHandler(req: MockRequest) {
+  switch (req.body?.token) {
+    case undefined:
+      return {
+        status: 200,
+        body: json({
+          pets: FirstPage,
+          nextToken: "12345",
+        }),
+      } as const;
+    case "12345":
+      return {
+        status: 200,
+        body: json({
+          pets: SecondPage,
+        }),
+      } as const;
+    default:
+      throw new ValidationError(
+        "Unsupported continuation token",
+        `"undefined" | "12345"`,
+        req.body.token,
+      );
+  }
+}
+
+Scenarios.Payload_Pageable_ServerDrivenPagination_ContinuationToken_reqBodyResBody = passOnSuccess([
+  {
+    uri: "/payload/pageable/server-driven-pagination/continuationtoken/request-body-response-body",
+    method: "get",
+    request: {},
+    response: EmptyResponse,
```

Comment:
Should we just put response of `FirstPage` here?

The mockApi has other purpose, so I'd assume we'd better to have some value response (instead of this `EmptyResponse` without body, that should not happen).

Reply:
Actual response is returned by `handler`. And as https://github.com/microsoft/typespec/pull/6038#discussion_r1959126179, `EmptyResponse` is just to pass build.